assets/js/base/components/payment-methods/payment-method-error-boundary.js

```diff
@@ -36,11 +36,14 @@ class PaymentMethodErrorBoundary extends Component {
 					);
 				}
 			}
-			return (
```

Comment:
Comment on the init of `let errorText =` in existing code:

```js
let errorText = __(
				'This site is experiencing difficulties with this payment method. Please contact the owner of the site for assistance.',
				'woo-gutenberg-products-block'
			);
```

Do we need the default/front end error message anymore? I'm not seeing it in the front end as shopper, so I'm assuming it's not used. 

Or to clarify the intent of this error boundary object .. it catches errors from individual payment methods, and if user is admin, shows an error notice inline about the problem with the payment method. 

Another comment on the existing code - `isEditor || CURRENT_USER_IS_ADMIN` - can we simplify this check?

Reply:
> Do we need the default/front end error message anymore? I'm not seeing it in the front end as shopper, so I'm assuming it's not used.

I think so. The error boundary can catch _any_ error thrown in the inner components, that means there might be unexpected errors that we can't anticipate with 3rd-party payment methods. The default error message allows the block to display a friendly message for shoppers.

> Another comment on the existing code - `isEditor || CURRENT_USER_IS_ADMIN` - can we simplify this check? 

What do you mean by simplify? Non-admins can also see the block in the editor, so in practice `isEditor` and `CURRENT_USER_IS_ADMIN` are different checks. I think it makes sense explicitly using both, but maybe I'm missing something? :thinking:

Reply:
Thanks for explaining about the default error message. I might test throwing a generic unexpected error to confirm the behaviour.

With the logic, I agree it probably doesn't need changing, that check is complete. I think I was worried about the "`isEditor` but can't change payment method settings" case.

Reply:
You mean a user that can see the block in the editor but doesn't have permissions to modify the payment methods settings? That's a good point, it might make sense having a separate error message for those cases. I would keep it for a follow-up, however, so we don't make this PR more complex than what already is. :slightly_smiling_face: